src/main/java/frc/robot/subsystems/CDSSubsystem.java

```diff
@@ -47,45 +53,140 @@
+  
+    } else {
       CDSWheelControllerOne.getSparkMax().set(Constants.CDSWheelControllerSpeed);
-      CDSWheelControllerDirection.setString("Reverse");
+      SmartDashboard.putString("CDS Wheel Direction", "Forward");
+      SmartDashboard.putNumber("CDS Wheel Speed", Constants.CDSWheelControllerSpeed);
+      
+    }
+  }
 
+  public void CDSBeltToggle(boolean reverse) {
+    if(reverse){
       CDSBeltController.getSparkMax().set(-Constants.CDSBeltSpeed);
       CDSBeltController.setIdleMode(IdleMode.kBrake);
-      CDSBeltControllerDirection.setString("Reverse");
+      SmartDashboard.putString("CDS Belt Direction", "Reverse");
+      SmartDashboard.putNumber("CDS Belt Speed", -Constants.CDSBeltSpeed);
 
     } else {
-      CDSWheelControllerOne.getSparkMax().set(Constants.CDSWheelControllerSpeed);
-      CDSWheelControllerDirection.setString("Forward");
-
-      CDSBeltController.getSparkMax().set(-Constants.CDSBeltSpeed);
-      CDSBeltControllerDirection.setString("Forward");
+      CDSBeltController.getSparkMax().set(Constants.CDSBeltSpeed);
+      SmartDashboard.putString("CDS Belt Direction", "Forward");
+      SmartDashboard.putNumber("CDS Belt Speed", Constants.CDSBeltSpeed);
     }
   }
 
-  public int[] getSensorStatus() {
-    int frontStatus = colorSensorOne.getProximity() > 800 ? 1 : 0;
-    int backStatus = backBeamBreak.get() ? 1 : 0;
-    int[] beamBreakArray = {frontStatus, backStatus};
+  public double getBeltSpeed() {
+    return CDSBeltController.getSpeed();
+  }
+
+  public double getWheelSpeed() {
+    return CDSWheelControllerOne.getSpeed();
+  }
+
+  public void stopCDS() {
+    // stops all motors in the CDS
+    CDSWheelControllerOne.getSparkMax().set(0.0);
+    CDSBeltController.getSparkMax().set(0.0);
+    SmartDashboard.putNumber("CDS Wheel Speed", 0.0);
+    SmartDashboard.putNumber("CDS Belt Speed", 0.0);
+  }
+
+  public void stopCDSWheel() {
+    // Stops only the centering wheels
+    CDSWheelControllerOne.getSparkMax().set(0.0);
+    SmartDashboard.putNumber("CDS Wheel Speed", 0.0);
+  }
+  
+  public boolean[] getSensorStatus() {
+    int[] sensorStatuses = colorSensors.getProximities();
+
+    boolean backStatus = sensorStatuses[0] > Constants.backSensorActivation;
+    boolean middleStatus = sensorStatuses[1] > Constants.middleSensorActivation;
+    boolean frontStatus = sensorStatuses[2] > Constants.frontSensorActivation;
+    boolean[] beamBreakArray = {backStatus, middleStatus, frontStatus};
+
+    int ballCount = 0;
+    for (boolean status : beamBreakArray) {
+      if (status) {
+        ballCount++;
+      }
+    }
+
+    SmartDashboard.putNumber("Ball Count", ballCount);
+
     return beamBreakArray;
   }
 
+  public int getNextOpenSensor(boolean[] sensorSta) {
+
+    boolean[] sensorStatus = getSensorStatus();
+    
+    // Start at 1 to skip over the centering wheel status 
+    for (int i=0; i < sensorStatus.length-1; i++) {
```

Comment:
Doesn't start at 1 like your comment suggests, could return 0

Reply:
Don't know how that got changed. I'll fix that now.

Reply:
Turns out the comment was outdated but the code was solid. I changed the comment to reflect the current state of things.